src/jdk.incubator.foreign/share/classes/jdk/internal/foreign/LayoutPath.java

```diff
@@ -185,6 +192,27 @@ public VarHandle dereferenceHandle(Class<?> carrier) {
         return handle;
     }
 
+    private static long addScaledOffset(long base, long index, long stride) {
+        return base + (stride * index);
+    }
+
+    public MethodHandle offsetHandle() {
+        MethodHandle mh = MethodHandles.identity(long.class);
+        LayoutPath cur = this;
+        while (cur.enclosing != null) {
+            if (cur.enclosing.layout instanceof SequenceLayout
+                && cur.elementIndex == UNSPECIFIED_ELEM_INDEX) { // sequenceElement()
+                MethodHandle collector = MethodHandles.insertArguments(MH_ADD_SCALED_OFFSET, 2, cur.layout.bitSize());
```

Comment:
You already have the `strides` array for doing this? E.g. I don't think there's a need to walk the layout paths upwards and to get their size - all you need is in the stride array.

Reply:
We need to walk to inspect the element indices. Could replace the `cur.layout.bitSize` with a lookup into the strides array maybe, but I think that would require keeping track of a separate index as well. (I'll take a look).

Reply:
My recollection of this code is that when you get to generating the handle, you already have the unspecified paths in the stride array. Look at the places for when a new stride is set - I think it should match the places where you need the info as well (that is `PathElement::sequenceElement()` and `PathElement::sequenceElement(long, long)`).

Reply:
Ah, right. The strides array only contains elements for unspecified indexes, not all sequence layouts. I'm confusing myself